auth/README.md

```diff
@@ -0,0 +1,364 @@
+
+```
+
+
+## Advanced Authentication
+
+### Create a Vimeo Account via Email
+
+If you would like to create a Vimeo account via email, you will need to provide the display name, email and password, and whether you want to opt out of marketing emails. This can be done as follows:
+
+Kotlin
+
+```kotlin
+
+// Kotlin
+authenticator.emailJoin(
+        display_name = "John Doe", 
+        email = "john.doe@gmail.com", 
+        password = "password", 
+        marketingOptIn = false, 
+        callback = object : VimeoCallback<AuthenticatedAccessToken> {
+              
+               override fun onSuccess(response: VimeoResponse.Success<AuthenticatedAccessToken>) { ...  }
+
+               override fun onError(error: VimeoResponse.Error) { ... }
+
+        })
+            
+```
+
+```java
+
+// Java
+authenticator.emailJoin(
+        "John Doe", 
+        "john.doe@gmail.com", 
+        "password", 
+        false,
+        new VimeoCallback<AuthenticatedAccessToken>() {
+                    
+            @Override
+            public void onSuccess(@NotNull Success<AuthenticatedAccessToken> response) { }
+
+            @Override
+            public void onError(@NotNull Error error) { }
+            
+        });
+
+```
+
+Upon success, you will get a [AuthenticatedAccessToken](../models/src/main/java/com/vimeo/networking2/AuthenticatedAccessToken.kt) which contains the access token and the authenticated [User](../models/src/main/java/com/vimeo/networking2/User.kt) object.
+
+### Login via email
+
+```kotlin
+
+// Kotlin
+authenticator.emailLogin(
+        email = "john.doe@gmail.com",
+        password = "password",
+        authCallback = object: VimeoCallback<AuthenticatedAccessToken> {
+                  
+             override fun onSuccess(response: VimeoResponse.Success<AuthenticatedAccessToken>) { ... }
+             
+             override fun onError(error: VimeoResponse.Error) { ... }
+
+        })
+            
+```
+
+```java
+
+// Java
+authenticator.emailLogin(
+        "john.doe@gmail.com", 
+        "password", 
+        new VimeoCallback<AuthenticatedAccessToken>() {
+            
+            @Override
+            public void onSuccess(@NotNull Success<AuthenticatedAccessToken> response) { }
+
+            @Override
+            public void onError(@NotNull Error error) { }
+        });
+
+```
+
+### Facebook and Google Authentication
+
+There is a single method the SDK provides to create a Vimeo Account with Google or Facebook or to login via these social channels. 
+
+```kotlin
+
+// Kotlin
+authenticator.google(
+        token = "google_token",
+        email = "john.doe@gmail.com",
+        marketingOptIn = false,
+        authCallback = object : VimeoCallback<AuthenticatedAccessToken> {
+                    
+            override fun onSuccess(response: VimeoResponse.Success<AuthenticatedAccessToken>){ ... }
+
+            override fun onError(error: VimeoResponse.Error) { ... }
```

Comment:
In the examples, inconsistent spacing before/after the `onSuccess` and `onError` methods is used. We should be consistent.